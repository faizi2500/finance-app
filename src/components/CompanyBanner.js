import React from 'react';
import PropTypes from 'prop-types';

const CompanyBanner = ({
  name,
  image,
  exchange,
  sector,
  price,
  description,
  volAvg,
  website,
}) => {
  const newdes = description.split(/\s+/).slice(0, 35).join(' ');
  return (
    <div className="parent-banner-div">
      <div className="banner-div">
        <div className="banner-img">
          <img src={image} alt="company logo N/A" />
        </div>
        <div className="banner-details">
          <h1>{name}</h1>
          <p>
            Exchange:
            {' '}
            &quot;
            <strong>
              {exchange}
            </strong>
            &quot;
          </p>
          <p>
            Sector:
            {' '}
            &quot;
            <strong>
              {sector}
            </strong>
            &quot;
          </p>
          <p>
            Price:
            {' '}
            &quot;
            <strong>
              {price}
            </strong>
            &quot;
          </p>
          <p>
            Average Volume:
            {' '}
            &quot;
            <strong>
              {volAvg}
            </strong>
            &quot;
          </p>
        </div>
      </div>
      <div className="banner-description-para">
        <p>
          {newdes}
          ....
          {' '}
          <br />
          (
          More at
          {' '}
          <a href={website} className="web">
            {website}
          </a>
          {' '}
          )
        </p>
      </div>
    </div>
  );
};

CompanyBanner.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  exchange: PropTypes.string.isRequired,
  sector: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  volAvg: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired,
};

export default CompanyBanner;
