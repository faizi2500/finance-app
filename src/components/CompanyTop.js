import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './company.css';
import { FaMicrophone, FaCog, FaArrowLeft } from 'react-icons/fa';

const CompanyTop = ({ symbol }) => (
  <div className="top-banner-parent">
    <div className="flex-company-banner">
      <NavLink to="/">
        <FaArrowLeft />
      </NavLink>
      <h4 className="navTitle">{symbol}</h4>
      <div className="right-side-banner">
        <FaMicrophone />
        <FaCog />
      </div>
    </div>
  </div>
);

CompanyTop.propTypes = {
  symbol: PropTypes.string.isRequired,
};

export default CompanyTop;
