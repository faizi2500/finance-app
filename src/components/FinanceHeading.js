import React from 'react';
import PropTypes from 'prop-types';

const FinanceHeading = ({ symbol }) => (
  <div className="finance-heading-container">
    <div className="finance-heading">
      {symbol}
      {' '}
      -
      {' '}
      Financial Indicators
    </div>
  </div>
);

FinanceHeading.propTypes = {
  symbol: PropTypes.string.isRequired,
};

export default FinanceHeading;
