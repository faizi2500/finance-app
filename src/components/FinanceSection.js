import React from 'react';
import PropTypes from 'prop-types';

const FinanceSection = ({ eachObj, indexValue }) => {
  const name = Object.keys(eachObj);
  const stringified = JSON.stringify(name);
  const splicedData = stringified.slice(0, -5);
  const finalIndicator = splicedData.charAt(2).toUpperCase() + splicedData.slice(3);
  const value = Object.values(eachObj);
  const checkValue = (!value[0]) ? 'N/A' : value;

  const nameOfClass = (indexValue % 2) ? 'evenFinancial' : 'oddFinancial';
  return (
    <div className={nameOfClass}>
      <div className="financialDes">
        <h3>{finalIndicator}</h3>
        <p>{checkValue}</p>
      </div>
    </div>
  );
};

FinanceSection.propTypes = {
  eachObj: PropTypes.instanceOf(Object).isRequired,
  indexValue: PropTypes.number.isRequired,
};

export default FinanceSection;
