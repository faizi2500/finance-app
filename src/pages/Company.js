import React, { useEffect, useState } from 'react';
// import { BrowserRouter as Router } from 'react-router-dom';
import { useLocation } from 'react-router';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCompanyFinancials } from '../redux/company/company';
import CompanyPage from '../components/CompanyPage';
import '../components/company.css';

const Company = () => {
  const data = useLocation();
  const url = window.location.href;
  const newString = url.substring(39);
  let symbol = newString;
  const dispatch = useDispatch();
  useEffect(() => {
    if (symbol) {
      dispatch(fetchCompanyFinancials(symbol));
    }
  }, [symbol]);
  const companyData = useSelector((state) => state.companyReducer.payload);
  const [newState, setState] = useState(companyData);
  if (data.name) {
    setState('');
  }
  setTimeout(() => {
    setState(companyData);
  }, 2000);
  return (
    <>
      <CompanyPage info={newState} symbol={symbol} />
    </>
  );
};

export default Company;
