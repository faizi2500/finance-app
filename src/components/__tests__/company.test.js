import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Route } from 'react-router-dom';
import store from '../../redux/configureStore';
import Company from '../../pages/Company';
import CompanyTop from '../CompanyTop';
import CompanyBanner from '../CompanyBanner';
import FinanceHeading from '../FinanceHeading';
import FinanceSection from '../FinanceSection';

test('Second company page render', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <Company />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

test('company page banner render', () => {
  const price = 25;
  const vol = 12452;
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <CompanyBanner
            name="Metal Steel"
            image="https://financialmodelingprep.com/image-stock/GRVI.png"
            exchange="NASDAQ"
            sector="Steel"
            price={price}
            description="Steel based company"
            volAvg={vol}
            website="https://mahroze.com/"
          />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

test('company page financial section render', () => {
  const financialObj = {
    priceEarningsRatioTTM: 1,
  };
  let onlyObj;
  Object.entries(financialObj).map((each) => {
    let keyIndex = '';
    let valueIndex = 0;
    each.forEach((val, index) => {
      if (index === 0) {
        keyIndex = val;
      }
      valueIndex = val;
    });
    onlyObj = {
      [keyIndex]: valueIndex,
    };
    return onlyObj;
  });
  const i = 0 + 1;
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <FinanceSection eachObj={onlyObj} indexValue={i} />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

test('company page heading render', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <FinanceHeading symbol="AAPL" />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

test('Each company page render', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <CompanyTop symbol="AAPL" />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});
