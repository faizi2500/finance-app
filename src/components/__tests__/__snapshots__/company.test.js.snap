// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Each company page render 1`] = `
<div
  className="top-banner-parent"
>
  <div
    className="flex-company-banner"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
        />
      </svg>
    </a>
    <h4
      className="navTitle"
    >
      AAPL
    </h4>
    <div
      className="right-side-banner"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Second company page render 1`] = `
Array [
  <div
    className="top-banner-parent"
  >
    <div
      className="flex-company-banner"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          />
        </svg>
      </a>
      <h4
        className="navTitle"
      >
        
      </h4>
      <div
        className="right-side-banner"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <div
    className="beingLoaded"
  >
    <svg
      className="rotateSpinner"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      />
    </svg>
  </div>,
]
`;

exports[`company page banner render 1`] = `
<div
  className="parent-banner-div"
>
  <div
    className="banner-div"
  >
    <div
      className="banner-img"
    >
      <img
        alt="company logo"
        src="https://financialmodelingprep.com/image-stock/GRVI.png"
      />
    </div>
    <div
      className="banner-details"
    >
      <h1>
        Metal Steel
      </h1>
      <p>
        Exchange:
         
        "
        <strong>
          NASDAQ
        </strong>
        "
      </p>
      <p>
        Sector:
         
        "
        <strong>
          Steel
        </strong>
        "
      </p>
      <p>
        Price:
         
        "
        <strong>
          25
        </strong>
        "
      </p>
      <p>
        Average Volume:
         
        "
        <strong>
          12452
        </strong>
        "
      </p>
    </div>
  </div>
  <div
    className="banner-description-para"
  >
    <p>
      Steel based company
      ....
       
      <br />
      ( More at
       
      <a
        className="web"
        href="https://mahroze.com/"
      >
        https://mahroze.com/
      </a>
       
      )
    </p>
  </div>
</div>
`;

exports[`company page financial section render 1`] = `
<div
  className="evenFinancial"
>
  <div
    className="financialDes"
  >
    <h3>
      PriceEarningsRatio
    </h3>
    <p>
      1
    </p>
  </div>
</div>
`;

exports[`company page heading render 1`] = `
<div
  className="finance-heading-container"
>
  <div
    className="finance-heading"
  >
    AAPL
     
    -
     
    Financial Indicators
  </div>
</div>
`;
